src/select/select.js

```diff
@@ -189,21 +189,24 @@ class Select extends React.Component<PropsT, StatelessStateT> {
         SearchIcon: SearchIconOverride,
       } = {},
     } = this.props;
+
     const Root = getOverride(RootOverride) || StyledRoot;
     const Input = getOverride(InputOverride) || StyledInput;
     const SearchIcon = getOverride(SearchIconOverride) || StyledSearchIcon;
-    const {label, placeholder} = this.props;
+    const {label, placeholder, disabled} = this.props;
     const {selectedOptions} = this.state;
     return (
       <div
         onClick={() => {
-          this.setState({isDropDownOpen: !this.state.isDropDownOpen});
+          if (!disabled) {
+            this.setState({isDropDownOpen: !this.state.isDropDownOpen});
+          }
         }}
       >
         <InputComponent
-          disabled={true}
```

Comment:
if we don't set it to true, we will be able to input symbols in Select mode, which is not what expected by design. Only in Search mode it should be available.

Reply:
what do you recommend, what we should do here?

Reply:
so setting it to disabled was intentional as we reuse Input component and it takes care of disabled state. So functionally it is correct. Unfortunately it causes a change in styles, which we don't have in Select by design. So I suggest to have disabled=true, but override styles of each subcomponent of Input by adding something like this:
```
overrides: {
        Input: InputOverride,
      },
    } = this.props;

InputOverride = Object.assign(InputOverride, {
    style: {
        //styles to override for disabled state of Input
     }
});

        <Input
          {...getOverrideProps(InputOverride)}
        >
```

Reply:
I am not sure if I understand why do we need to set disabled to true on the initial render - can you help me what am I missing?

Reply:
In Select mode we are not allowed to input anything, only display. So we set Input component to disabled to avoid any work with keyboard

Reply:
Can we just use a `div` instead of an Input for select mode? Seems like the input is only needed for search mode.

Reply:
I tried. However, input has more than half of styles needed for Select. It's not just a div, but some subcomponents. It seems obvious not to DRY and reuse Input , which already supports styles and subcomponents. Otherwise we need to repeat all of it.

Reply:
|   |   |
|---|---|
| <img width="347" alt="screen shot 2018-09-06 at 11 58 44 am" src="https://user-images.githubusercontent.com/875591/45179376-31fee300-b1cd-11e8-95d9-5d0e8a2da744.png"> | <img width="379" alt="screen shot 2018-09-06 at 11 59 00 am" src="https://user-images.githubusercontent.com/875591/45179380-33c8a680-b1cd-11e8-8699-d4d0cd097e17.png">  |

So they look similar, but how many styles do they actually share? Maybe the padding, typography, and border radius? Repeating those styles doesn't sound that bad, especially if it removes a whole class of random bugs that crop up (like this `disabled` one) by trying too hard to be 100% DRY.

Reply:
styles:
padding, fonts, background color, border
functional:
Before and After icons and their styles, Placeholder and selected value
So if we wish it separate then for Select mode the same functional part should be repeated and maintained outside of Input and transferred to Select itself

Reply:
I think we need to rethink the select implementation here since using a disabled input feels wrong semantically-, visually-, and accessibility-wise